assets/src/edit-story/components/form/dateTime/timePicker.js

```diff
@@ -0,0 +1,367 @@
+    ) {
+      return;
+    }
+
+    const newDate = is12Hour
+      ? date.clone().hours(am === 'AM' ? value % 12 : ((value % 12) + 12) % 24)
+      : date.clone().hours(value);
+    changeDate(newDate);
+  };
+
+  const updateDay = () => {
+    const { day, date } = state;
+    const value = parseInt(day);
+    if (isNaN(value) || value < 1 || value > 31 || day === value) {
+      return;
+    }
+    const newDate = date.clone().date(value);
+    changeDate(newDate);
+  };
+
+  const updateMonth = () => {
+    const { month, date } = state;
+    const value = parseInt(month);
+    if (isNaN(value) || value < 1 || value > 12 || month === value) {
+      return;
+    }
+    const newDate = date.clone().month(value - 1);
+    changeDate(newDate);
+  };
+
+  const updateYear = () => {
+    const { year, date } = state;
+    const value = parseInt(year);
+    if (isNaN(value) || value < 0 || value > 9999 || year === value) {
+      return;
+    }
+    const newDate = date.clone().year(value);
+    changeDate(newDate);
+  };
+
+  const updateAmPm = (value) => () => {
+    const { am, date, hours } = state;
+    if (am === value) {
+      return;
+    }
+    let newDate;
+    if (value === 'PM') {
+      newDate = date.clone().hours(((parseInt(hours) % 12) + 12) % 24);
+    } else {
+      newDate = date.clone().hours(parseInt(hours) % 12);
+    }
+    changeDate(newDate, { am: value });
+  };
+
+  const renderDay = (day) => {
+    return (
+      <div key="render-day">
+        <NumberInput
+          aria-label={__('Day', 'web-stories')}
+          type="number"
+          value={day}
+          step={1}
+          min={1}
+          onChange={onChangeEvent('day')}
+          onBlur={updateDay}
+          width={35}
+        />
+      </div>
+    );
+  };
+
+  const renderMonth = (month) => {
+    return (
+      <div key="render-month">
+        <SelectInput
+          aria-label={__('Month', 'web-stories')}
+          value={month}
+          onChange={onChangeEvent('month')}
+          onBlur={updateMonth}
+        >
+          <option value="01">{__('January', 'web-stories')}</option>
+          <option value="02">{__('February', 'web-stories')}</option>
+          <option value="03">{__('March', 'web-stories')}</option>
+          <option value="04">{__('April', 'web-stories')}</option>
+          <option value="05">{__('May', 'web-stories')}</option>
+          <option value="06">{__('June', 'web-stories')}</option>
+          <option value="07">{__('July', 'web-stories')}</option>
+          <option value="08">{__('August', 'web-stories')}</option>
+          <option value="09">{__('September', 'web-stories')}</option>
+          <option value="10">{__('October', 'web-stories')}</option>
+          <option value="11">{__('November', 'web-stories')}</option>
+          <option value="12">{__('December', 'web-stories')}</option>
+        </SelectInput>
+      </div>
+    );
+  };
+
+  const renderDayMonthFormat = (is12HourFormat) => {
```

Comment:
Technically, it's not the same localisation setting, that says 12 hour format and big-endian date format. Egypt for instance (like most of NE Africa) uses little-endian date format (DMY) but 12 hour time format.

Compare [this map of date formats](https://en.wikipedia.org/wiki/Date_format_by_country) with [this map of time formats](https://en.wikipedia.org/wiki/Date_and_time_representation_by_country#/media/File:12_24_Hours_World_Map.svg).